src/main/java/dansplugins/factionsystem/commands/CreateCommand.java

```diff
@@ -11,8 +12,58 @@
 import org.bukkit.command.CommandSender;
 import org.bukkit.entity.Player;
 
-public class CreateCommand {
+public class CreateCommand extends SubCommand {
 
+    public CreateCommand() {
+        super(new String[] {
+                LOCALE_PREFIX + "CmdCreate", "Create"
+        }, true);
+    }
+
+    /**
+     * Method to execute the command for a player.
+     *
+     * @param player who sent the command.
+     * @param args   of the command.
+     * @param key    of the sub-command (e.g. Ally).
+     */
+    @Override
+    public void execute(Player player, String[] args, String key) {
+        final String permission = "mf.create";
+        if (!(checkPermissions(player, permission))) return;
+        this.faction = getPlayerFaction(player);
+        if (this.faction != null) {
+            player.sendMessage(translate("&c" + getText("AlreadyInFaction")));
+            return;
+        }
+        if (args.length == 0) {
+            player.sendMessage(translate("&c" + getText("UsageCreate")));
+            return;
+        }
```

Comment:
It should be fairly simple to fix the problem mentioned above by adding the following code segment to the checks that happen before the creation:


```
if (PersistentData.getInstance().getFaction(name) != null) {
    player.sendMessage(ChatColor.RED + LocaleManager.getInstance().getText("FactionAlreadyExists"));
    return false;
}
```

If you want, I can even do it after merging the pull request.

Reply:
I've decided to handle this afterwards.